File: python/n414_ThirdNum.py
# https://leetcode-cn.com/problems/third-maximum-number/submissions/
class Solution:
    def thirdMax(self, nums) -> int:
        if not nums: return None
        tmp = list(set(nums))
        tmp.sort()
        if len(tmp) >= 3: return tmp[-3]
        else: return tmp[-1]
        
if __name__ == "__main__":
    solu = Solution()
    nums = [2, 5, 3, 8, 5, 5, 8, 8, 4, 12]
    print(solu.thirdMax(nums))

File: cpp/n414_ThirdNum.cpp
// https://leetcode-cn.com/problems/third-maximum-number/submissions/
#include <iostream>
#include <vector>
#include <string>
#include <set>
#include <algorithm>
using namespace std;

class Solution{
public:
    int thirdMax(vector<int>& nums){
        set<int> s (nums.begin(), nums.end());
        auto it = s.end();
        it--;
        if(s.size() >= 3){
            it--;
            it--;
        }
        return *it;
    }
};

int main(){
    return 0;
}